### PR TITLE
Add additional logging for debugging docker and localstack

### PR DESCRIPTION
**Pull request by DavidTanner**
debug

Sometimes the build is paused, and it is nice to be able to see what is going on.  Specifically, debugging the recent issues with Elasticsearch on localstack needed the container logs to see what was going on.

### CODE REVIEW

File: src/docker.js [hunk 1]
```diff
@@ -62,8 +62,14 @@ const getInterfaceAddress = (ifconfig) => {
 };
 
 const pullImage = async (docker, image) => {
-  const followProgress = promisify(docker.modem.followProgress);
-  await followProgress(await docker.pull(image));
+  const stream = await docker.pull(image);
+  await new Promise(async (resolve, reject) => {
+    docker.modem.followProgress(stream, resolve, (progress) => {
+      if (debugDocker) {
+        console.log(`${image}: ${progress.status} ${progress.progress ? progress.progress : ''}`);
```
Comment (Druotic) [inline src/docker.js]: Doesn't this just print progress on pulling the image? Based on the docs ('To debug some docker logs...'), I would have likely guessed runtime docker container logs. `docker logs` is a docker CLI command which prints container logs, so making this assumption seems reasonable to someone looking at the docs only. Perhaps a tweak to the description would make it more clear.

Since this is the third (unless I missed some) place where we conditionally enable debug logging (log namespacing more or less), perhaps it would be a good time to switch to a tool that does just this rather than chaining a bunch of special purpose env vars. Perhaps [`debug`](https://github.com/visionmedia/debug) or some other similarly popular library.  Enabling logging would simply become `DEBUG=lambda,docker,localstack yarn test`

Not a blocker, but I think it'd simplify things a bit and set us up nicely for future additions. Especially if you think about having to forward all of these environment variables vs. just one (potentially multiple times if nesting containers).

File: src/localstack.js [hunk 2]
```diff
@@ -210,12 +218,12 @@ function checkServices (services = []) {
...
   const stream = await container.attach({ stream: true, stdout: true, stderr: true });
   return new Promise((resolve) => {
-    const logs = new TempWriteBuffer(resolve);
+    const logs = new TempWriteBuffer(resolve, container.id);
     container.modem.demuxStream(stream, logs, logs);
   });
 }
 
-async function getConnection ({ versionTag = 'latest', services } = {}) {
+async function getConnection ({ versionTag = '0.10.6', services } = {}) {
```
Comment (Druotic) [inline src/localstack.js]: This seems unrelated to the title. What was the motivation for this? Defaulting to 'latest' feels pretty natural for a lot of other systems (e.g. pulling a docker image with no version specified).

Comment (DavidTanner) [reply src/localstack.js]: Over the last month the latest image hasn't been stable.  It also didn't match any official release.  Using latest is dangerous as it can cause previously passing builds to fail, or even cause issues in production.  
`:latest` does not mean the latest version, just whatever was added to the repo.  This can be older versions, that may or may not be compatible with current requirements.

https://vsupalov.com/docker-latest-tag/
https://medium.com/@mccode/the-misunderstood-docker-tag-latest-af3babfd6375
https://blog.container-solutions.com/tagging-docker-images-the-right-way

File: src/docker.js (hunk 1 above)
Comment (DavidTanner) [reply src/docker.js]: I like the idea of replacing all of the console logs in the project to instead use some other library.  I am thinking Bunyan would be appropriate here, as we already use it for LifeOmic and it formats logs nicely.  The only other part would be parsing the log level from the env, or allowing it to be set in the library.

File: README.md [hunk 3]
```diff
@@ -358,3 +358,8 @@ You will also find the following intermediate files:
 
  The `WEBPACK_MODE=development` environment variable will prevent
  minification in the final output bundle.
+
+## Debugging
+
+To debug some docker logs, set `DEBUG_DOCKER=true`.
+To debug the localstack docker container startup, set `DEBUG_LOCALSTACK=true`.
```
[the remark this answers is not in the document]
Comment (DavidTanner) [reply README.md]: I'll update this as well as the other logging.

File: src/localstack.js (hunk 2 above)
Comment (Druotic) [reply src/localstack.js]: I agree that using 'latest' is generally a bad idea - particularly for reproducibility reasons. However, if this hardcoded semi-arbitrary version doesn't get updated regularly, as a consumer, I might get bitten by hard to track down bugs (i.e. old version not supporting something) due to using a version I never explicitly specified. I suspect I might would assume the latest version was being used until I had reason to believe otherwise.

At any rate, perhaps documenting this in the README somewhere would be beneficial.

Comment (DavidTanner) [reply src/localstack.js]: Users of the library can specify their own version.  Either way brings risk, but at least if we pick a version we can know what the risks are, instead of getting random errors that could be impossible to track down using the latest tag. 

I will update the README
